tcms/testcases/tests/test_views.py

```diff
@@ -228,6 +232,71 @@ def _assertTestCase(self, test_case):
         self.assertEqual(test_case.notes, self.notes)
 
 
+class TestNewCasePermission(PermissionsTestCase):
+    permission_label = 'testcases.add_testcase'
+    http_method_names = ['get', 'post']
+    url = reverse('testcases-new')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.summary = 'summary'
+        cls.post_data = {
+            'summary': cls.summary,
+        }
+        super().setUpTestData()
+        user_should_have_perm(cls.tester, 'testcases.view_testcase')
+
+        case_status_confirmed = TestCaseStatus.get_confirmed()
+
+        case = TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=case_status_confirmed,
+        )
+        case.save()  # creates a new case entry in the database
+
+        cls.post_data.update({
+            'author': cls.tester.pk,
+            'default_tester': cls.tester.pk,
+            'priority': case.priority.pk,
+            'product': case.category.product.pk,
+            'category': case.category.pk,
+            'case_status': case_status_confirmed.pk,
+            'text': 'some text description',
+            'script': 'some script',
+            'arguments': 'args1, args2, args3',
+            'email_settings-0-auto_to_case_author': 'on',
+            'email_settings-0-auto_to_run_manager': 'on',
+            'email_settings-0-auto_to_case_run_assignee': 'on',
+            'email_settings-0-auto_to_case_tester': 'on',
+            'email_settings-0-auto_to_run_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_case_delete ': 'on',
+            'email_settings-0-cc_list': 'info@example.com',
+            'email_settings-0-case': '',
+            'email_settings-0-id': case.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        })
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _('New TestCase'))
+
+    def verify_post_with_permission(self):
+        response = self.client.post(self.url, self.post_data, follow=True)
+        test_case = TestCase.objects.get(summary=self.summary)
+        redirect_url = reverse('testcases-get', args=[test_case.pk])
+
+        self.assertRedirects(response, redirect_url, status_code=302, target_status_code=200)
+        self.assertEqual(test_case.summary, self.summary)
```

Comment:
This is equivalent to line 293. Search for self.summary or test_case.summary inside the response.

Reply:
Done